src/views/portals/HomeView.vue

```diff
@@ -0,0 +1,213 @@
+            <span v-if="$user.isLogged">
+              <input
+                id="c2c-personal-feed"
+                :class="{ 'switch is-rtl is-rounded': !$screen.isMobile }"
+                type="checkbox"
+                v-model="isPersonal"
+                @click="toogleProperty('isPersonal')"
+              />
+              <label
+                for="c2c-personal-feed"
+                :title="isPersonal ? $gettext('Personal feed on') : $gettext('Personal feed off')"
+              >
+                <span class="is-hidden-mobile" v-translate>Load my preferences</span>
+              </label>
+            </span>
+            <router-link to="preferences" class="has-text-normal" :title="$gettext('My preferences')">
+              <fa-icon icon="gears" />
+            </router-link>
+          </span>
+        </div>
+        <div class="feed-view" v-if="feed">
+          <feed-widget :type="isPersonal && $user.isLogged ? 'personal' : 'default'" hide-empty-documents />
+        </div>
+        <div v-if="!feed">
+          <dashboard-outings-list :is-personal="isPersonal" />
+          <dashboard-images-gallery v-if="$screen.isMobile" />
+          <dashboard-routes-list />
+          <dashboard-articles-list v-if="$screen.isMobile" />
+          <useful-links v-if="$screen.isMobile" />
+          <forum-widget :message-count="20" v-if="$screen.isMobile" />
+        </div>
+      </div>
+      <div
+        v-if="!$screen.isMobile"
+        class="column"
+        :class="feed ? 'is-5-tablet is-5-desktop is-4-widescreen is-3-fullhd' : 'is-5 is-4-fullhd'"
+      >
+        <publi-widget />
+        <dashboard-images-gallery v-if="!feed" />
+        <useful-links />
+        <forum-widget :message-count="20" />
+        <dashboard-articles-list v-if="!feed" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import HomeBanner from './HomeBanner';
+import DashboardArticlesList from './utils/DashboardArticlesList';
+import DashboardImagesGallery from './utils/DashboardImagesGallery';
+import DashboardOutingsList from './utils/DashboardOutingsList';
+import DashboardRoutesList from './utils/DashboardRoutesList';
+import ForumWidget from './utils/ForumWidget';
+import PubliWidget from './utils/PubliWidget';
+import UsefulLinks from './utils/UsefulLinks';
+
+import FeedWidget from '@/components/feed-widget/FeedWidget';
+
+export default {
+  name: 'HomeView',
+
+  components: {
+    HomeBanner,
+    DashboardArticlesList,
+    DashboardImagesGallery,
+    DashboardOutingsList,
+    DashboardRoutesList,
+    ForumWidget,
+    PubliWidget,
+    UsefulLinks,
+    FeedWidget,
+  },
+
+  data() {
+    let state = true;
+
+    if (this.$user.isLogged) {
+      state = false;
+    }
+    return {
+      isPersonal: this.$localStorage.get('isPersonal', false),
+      feed: this.$localStorage.get('feed', false),
+      visible: state,
+    };
+  },
+
+  methods: {
+    toogleProperty(property) {
```

Comment:
this is a bit to "open" for me, you could set any value. Are there so many properties we need to have a generic method for it?

Reply:
It's used for 3 different properties.